File: src/Sample.NativeLibrary/rectangle.cpp
#include <iostream>
#include "rectangle.h"
using namespace std;

Rectangle::Rectangle() = default;

Rectangle::Rectangle(const Point min, const Point max)
{
	this->Min = min;
	this->Max = max;
}

Rectangle::~Rectangle()
{
	std::cout << "C++ Rectangle析构函数调用" << std::endl;
}

File: src/Sample.Invoker/Models/Rectangle.cs
using System.Runtime.InteropServices;

namespace Sample.Invoker.Models
{
    [StructLayout(LayoutKind.Sequential)]
    public readonly struct Rectangle
    {
        public Rectangle(Point min, Point max)
        {
            this.Min = min;
            this.Max = max;
        }

        public readonly Point Min;
        public readonly Point Max;
    }
}
